core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -894,9 +894,11 @@ class ReplicaManager(val config: KafkaConfig,
                                requiredAcks: Short,
                                requestLocal: RequestLocal): Map[TopicPartition, LogAppendResult] = {
     val traceEnabled = isTraceEnabled
+    val failedTopicSet = mutable.Set[String]()
     def processFailedRecord(topicPartition: TopicPartition, t: Throwable) = {
       val logStartOffset = onlinePartition(topicPartition).map(_.logStartOffset).getOrElse(-1L)
-      brokerTopicStats.topicStats(topicPartition.topic).failedProduceRequestRate.mark()
+      // Every time appending to a local log fails, collect the topic names into the set.
+      failedTopicSet.add(topicPartition.topic)
       brokerTopicStats.allTopicsStats.failedProduceRequestRate.mark()
```

Comment:
we would want this metric to be incremented once per request instead of once per TP.
ditto for `totalProduceRequestRate`, `totalFetchRequestRate`, `failedFetchRequestRate`

Otherwise the `allTopicsStats` will be inconsistent with the sum of individual topic stats.

Reply:
(After some thinking) Oh, you are right. I omitted them. :sweat: